example-apps/internal-knowledge-search/app-ui/src/pages/SearchPage.tsx

```diff
@@ -0,0 +1,111 @@
+import SearchBar from "../components/SearchBar";
+import FilterBar from "../components/FilterBar";
+import SearchResults from "../components/SearchResults";
+import React, {useEffect, useState} from "react";
+
+import SearchApplicationClient from '@elastic/search-application-client';
+import {useDispatch, useSelector} from "react-redux";
+import {useToast} from "../contexts/ToastContext";
+import {updateSearchResults} from "../store/slices/searchResultsSlice";
+import SortBar from "../components/SortBar";
+import {MessageType} from "../components/Toast";
+import dataSourceToLogoPathLookup from '../config/dataSourceToLogoPathLookup.json';
+import documentsToSearchResultsMappings from '../config/documentsToSearchResultMappings.json';
+import {RootState} from "../store/store";
+
+
+const mapHitToSearchResult = (hit) => {
+    const doc = hit._source;
+    const dataSource = hit._index;
+    const dataSourceImagePath = dataSource in dataSourceToLogoPathLookup ? dataSourceToLogoPathLookup[dataSource] : "";
+    const dataSourceMapping = documentsToSearchResultsMappings[dataSource]
+
+    return {
+        title: doc[dataSourceMapping["title"]] || "No title",
+        created: doc[dataSourceMapping["created"]] || "No creation date",
+        previewText: doc[dataSourceMapping["previewText"]] || "No preview",
+        fullText: doc[dataSourceMapping["fullText"]] || "No full text",
+        dataSource: dataSource || "Data source unknown",
+        dataSourceImagePath: dataSourceImagePath || "",
+        link: doc[dataSourceMapping["link"]] || "Unknown"
+    };
+}
+
+export default function SearchPage() {
+    const dispatch = useDispatch();
+    const {showToast} = useToast();
+    const [loading, setLoading] = useState(false);
+    //TODO: simplify query state and move it to one place
+    const [query, setQuery] = useState<string>("");
+
+    const {appName, apiKey, searchEndpoint} = useSelector((state: RootState) => state.searchApplicationSettings);
+    const {sorts} = useSelector((state: RootState) => state.sort);
+    const indexFilter = useSelector((state: RootState) => state.filter)["filters"]["Data sources"].values;
+
+
+    useEffect(() => {
+        const fetchData = async () => await handleSearchSubmit();
+        fetchData();
+    }, [indexFilter, sorts, appName, apiKey, searchEndpoint]);
+
+    const handleSearchSubmit = async () => {
+        try {
+            setLoading(true);
+
+            const client = SearchApplicationClient(appName, searchEndpoint, apiKey, {
```

Comment:
As a future improvement u can initiate client when appName or searchEndpoint or apiKey change and here just call client()...
Every call of client initiates new query builder chain so it redundant to initiate new client every time on request

Reply:
I'm actually using this to workaround the built-in caching of the client as there's no way to disable the caching through the public API, right? So for demo purposes it's better to always show the actual newest results for a query instead of having to waiting for the internal caching time. Maybe we could expose these settings in the `search-application-client` through its public API?